portfolio/src/main/webapp/script.js

```diff
@@ -45,9 +50,216 @@ function getServerMessage() {
+              stylers: [{color: '#2f3948'}]
+            },
+            {
+              featureType: 'transit.station',
+              elementType: 'labels.text.fill',
+              stylers: [{color: '#d59563'}]
+            },
+            {
+              featureType: 'water',
+              elementType: 'geometry',
+              stylers: [{color: '#17263c'}]
+            },
+            {
+              featureType: 'water',
+              elementType: 'labels.text.fill',
+              stylers: [{color: '#515c6d'}]
+            },
+            {
+              featureType: 'water',
+              elementType: 'labels.text.stroke',
+              stylers: [{color: '#17263c'}]
+            }
+          ]
+        });
+
+    map.setTilt(45);
+
+    var contentMexico = '<div id="info-window">'+
+        '<h1>Xcaret Mexico</h1>'+
+        '<p><b>Xcaret</b> is one of greatest nature parks. '+
+        'Here you can find underground rivers, the caribbean '+
+        'and Mexico\'s culture.</p>'+
+        '<a href="https://www.xcaret.com/en/?langRedirect=1">Attribution: Xcaret by Mexico</a>'+
+        '</div>';
+
+    var contentChicago = '<div id="info-window">'+
+        '<h1>Museum of Science and Industry</h1>'+
+        '<p><b>Museum of Science and Industry in Chicago</b>, one of the largest '+
+        'museums in the world. It inspires people to pursue science careers and it'+
+        'is the home of more than 35,000 artifacts.</p>'+
+        '<a href="https://www.msichicago.org/">Attribution: Museum of Science and Industry</a>'+
+        '</div>';
+
+    var markerMexico = createMarkerForDisplay(20.5809134,-87.1218867,contentMexico);
+    var markerChicago = createMarkerForDisplay( 41.8028818,-87.5852623, contentChicago);
+
+    /** When the user clicks in the map, show a marker with a text box the user can edit. */
+    map.addListener('click',(event) => {
+        createMarkerForEdit(event.latLng.lat(), event.latLng.lng());
+    });
+
+    fetchMarkers();
+}
+
+/** Fetches markers from the backend and adds them to the map. */
+function fetchMarkers(){
+    fetch('/markers').then(response => response.json()).then((markers) =>{
+        markers.forEach((marker) => {
+            createMarkerForDisplay(marker.lat, marker.lng, marker.content);
+        });
+    });
+}
+
+/** Creates a marker that shows a read-only info window when clicked. */
+function createMarkerForDisplay(lat, lng, content) {
+  const marker = 
+    new google.maps.Marker({position: {lat: lat, lng: lng}, map: map});
+
+  var infowindow = new google.maps.InfoWindow({
+        content: content
+  });
+
+  marker.addListener('click',() => {
+        infowindow.open(map, marker);
+  });
+}
+
+/** Sends a marker to the backend for saving. */
+function postMarker(lat, lng, content) {
+  const params = new URLSearchParams();
+  params.append('lat', lat);
+  params.append('lng', lng);
+  params.append('content', content);
+
+  fetch('/markers', {method: 'POST', body: params});
```

Comment:
this would make a call and wait in the background until it times out. Better to have a success/failure message and log it in the results.